crates/hyperswitch_domain_models/src/payments/payment_attempt.rs

```diff
@@ -540,16 +544,24 @@ impl behaviour::Conversion for PaymentIntent {
             shipping_details: self.shipping_details.map(Encryption::from),
         })
     }
-
     async fn convert_back(
+        state: &KeyManagerState,
         storage_model: Self::DstType,
         key: &masking::Secret<Vec<u8>>,
+        key_store_ref_id: String,
     ) -> CustomResult<Self, ValidationError>
     where
         Self: Sized,
     {
         async {
-            let inner_decrypt = |inner| decrypt(inner, key.peek());
+            let inner_decrypt = |inner| {
+                decrypt(
```

Comment:
I don't think we have decrypt imported anywhere

Reply:
will be taken up in this PR.
https://github.com/juspay/hyperswitch/pull/5380